pydis_site/apps/api/viewsets/bot/whitelist.py

```diff
@@ -0,0 +1,69 @@
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import mixins
+from rest_framework.viewsets import GenericViewSet
+
+from pydis_site.apps.api.models.bot.whitelist import Whitelist
+from pydis_site.apps.api.serializers import WhitelistSerializer
+
+
+class WhitelistViewSet(
+    mixins.CreateModelMixin,
+    mixins.ListModelMixin,
+    mixins.DestroyModelMixin,
+    GenericViewSet,
+):
+    """
+    View providing create, list and  delete operations on whitelist shown by our bot.
+
+    ## Routes
+    ### GET /bot/whitelist?type=invites
```

Comment:
Don't include a query parameter as part of the route. Document the query parameters in a separate section.